tests/foreman/cli/test_user.py

```diff
@@ -295,20 +214,22 @@ def test_negative_create_with_invalid_username(self):
 
         @Assert: User is not created. Appropriate error shown.
         """
-        for invalid_name in ('',
-                             'space {0}'.format(gen_string('alpha')),
-                             gen_string('alpha', 101),
-                             gen_string('html')):
-            with self.subTest(invalid_name):
+        invalid_names = (
+            '', 'space {0}'.format(gen_string('alpha')),
+            gen_string('alpha', 101),
+            gen_string('html')
+        )
+        for invalid_name in invalid_names:
+            with self.subTest(invalid_name), self.assertRaises(
+                    CLIReturnCodeError):
```

Comment:
It woud be a problem only if FooError is subclass of CLIError. See:

```python
    def test_raise(self):
        with self.subTest('foo'):
            raise IndexError()
            with self.assertRaises(ValueError):
                pass

#Result:
self = <tests.robottelo.test_api_utils.UtilsTestCase testMethod=test_raise>

    def test_raise(self):
        with self.subTest('foo'):
>           raise IndexError()
E           IndexError

test_api_utils.py:25: IndexError
```

```python

    def test_raise(self):
        with self.subTest('foo'), self.assertRaises(ValueError):
            raise IndexError()


self = <tests.robottelo.test_api_utils.UtilsTestCase testMethod=test_raise>

    def test_raise(self):
        with self.subTest('foo'), self.assertRaises(ValueError):
>           raise IndexError()
E           IndexError

test_api_utils.py:25: IndexError
```

Probably assertRaises create the context manage and make the Exception assertion on its __exit__ method.